shared-libs/cht-datasource/src/local/libs/local-environment.ts

```diff
@@ -0,0 +1,40 @@
+import { getDocById } from '../../libs/doc';
+
+const SETTINGS_DOC_ID = 'settings';
+
+let settings: Doc | undefined;
+
+const getSettingsDoc = async (medicDb: PouchDB.Database<Doc>) => {
+  const settingsDoc = await getDocById(medicDb)(SETTINGS_DOC_ID);
+  if (!settingsDoc) {
+    throw new Error('Settings document not found.');
+  }
+
+  return settingsDoc;
+};
+
+const initFeed = (medicDb: PouchDB.Database<Doc>) => medicDb.changes({ since: 'now', live: true })
+  .on('change', async (change) => {
+    if (change.id === SETTINGS_DOC_ID) {
+      settings = await getSettingsDoc(medicDb);
+    }
+  });
+
+export interface LocalEnvironment {
+  medicDb: PouchDB.Database<Doc>;
+  getSettings: () => Doc;
+}
+
+export const getLocalEnvironment = async function(sourceDatabases?: SourceDatabases): Promise<Nullable<LocalEnvironment>> {
```

Comment:
```suggestion
export const getLocalEnvironment = async (sourceDatabases?: SourceDatabases): Promise<Nullable<LocalEnvironment>> => {
```

minor nitpick but let's stay consistent on the syntax